provider/kubernetes/kubernetes.go

```diff
@@ -270,8 +280,7 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 
 				service, exists, err := k8sClient.GetService(i.Namespace, pa.Backend.ServiceName)
 				if err != nil {
-					log.Errorf("Error while retrieving service information from k8s API %s/%s: %v", i.Namespace, pa.Backend.ServiceName, err)
-					return nil, err
+					return nil, fmt.Errorf("error while retrieving service information from k8s API %s/%s: %v", i.Namespace, pa.Backend.ServiceName, err)
```

Comment:
Is this change related to what the PR essentially tries to achieve?

I don't mind a good drive-by improvement. What I'm wondering here though is if we had a reason to log the error first. From what I can tell, [this line](https://github.com/dtomcej/traefik/blob/2248a79a27e8c71c8605e609d9b3ad2b84d9ea2f/provider/kubernetes/kubernetes.go#L159) will eventually log the error. I'm not sure if this reads well though as it would show up as

```
Cannot connect to Provider: error while retrieving service information from k8s API [...]
```

We are beyond the connection phase at this point already, aren't we?
How about annotating the [error returned here](https://github.com/dtomcej/traefik/blob/2248a79a27e8c71c8605e609d9b3ad2b84d9ea2f/provider/kubernetes/kubernetes.go#L137) like `return fmt.Errorf("failed to load Ingress: %s", err)` and log the error from line 159 as-in?

Similar for line 318.

Reply:
Looking at it in that perspective, I like the original logging, then return pattern.